build.sbt

```diff
@@ -152,17 +155,45 @@ lazy val connect = (project in file("graphframes-connect"))
     assembly / assemblyShadeRules := Seq(
       ShadeRule.rename("com.google.protobuf.**" -> protobufShadingPattern).inAll),
     assembly / assemblyMergeStrategy := {
-      case PathList("google", "protobuf", xs @ _*) => MergeStrategy.discard
       case PathList("META-INF", xs @ _*) => MergeStrategy.discard
       case x if x.endsWith("module-info.class") => MergeStrategy.discard
-      case x => MergeStrategy.first
+      case _ => MergeStrategy.first
     },
-    assembly / assemblyExcludedJars := (Compile / fullClasspath).value.filter { className =>
-      className.data
-        .getName()
-        .contains("scala-library-") || className.data
-        .getName()
-        .contains("slf4j-api-")
+    // I have zero ideas how to apply shading and exclude everything except connect classes in the right way.
+    // This looks terrible, but it works:
+    assembly / assemblyExcludedJars := (Compile / fullClasspath).value.filterNot { className =>
+      className.data.getName.contains("GraphFramesConnect") || className.data.getPath.contains(
+        "graphframes/connect/proto")
+    },
```

Comment:
I tried `sbt-shading` plugin but it is terrible. I was able to fix it with assembly.